File: javascript/str_join_slow.js
const chunk = "regexp123abcbcd";
const repeatCount = 10000;
const size = 1000;

function join() {
    const chunks = Array(size);
    for (let j = 0; j < size; ++j) chunks[j] = chunk;
    chunks.join("");
}

const start = Date.now();
for (let i = 0; i < repeatCount; ++i) {
    join();
}
const end = Date.now();

console.log(end - start);

File: ruby/str_join_slow.rb
chunk = "regexp123abcbcd"
repeat_count = 10000
size = 1000

def join(size, chunk)
  str = ""
  size.times { str += chunk }
end

start = Time.now
repeat_count.times { join(size, chunk) }
_end = Time.now

puts ((_end - start) * 1000).to_i
